### PR TITLE
Fix formatting of gas price for all conversion currencies

### LINKED ISSUE #3498: Some tokens when set as the "Conversion currency" not showing proper value
**Describe the bug**
Some currencies (tokens it looks like) when set as the “Conversion currency” shows “VALUE” + the abbreviation.

**Screenshots**
In these screenshots I had selected “PAY” and “BNT” as the “Conversion currency” in settings.
image
image

**To Reproduce**
1. Go to 'Settings > General'
2. Select 'BNT' as your 'Conversion currency'
3. Launch a transaction
4. See amount displayed as "VALUE BNT"

**Expected behavior**
It should show the value of the 'Conversion currency'.

**Smartphone (please complete the following information):**
 - Device: iPhone 12 Pro
 - OS: iOS 15.1.1
 - App Version: 3.7.1 (796)

-------------------------------------------------------------
_to be added after bug submission by internal support / PM_
**Severity**

- How critical is the impact of this bug on a user?
- Add stats if available on % of customers impacted
- Is this visible to all users?
- Is this tech debt?

### PR DESCRIPTION
**Pull request by Fatxx**
**Description**

Fix formatting of gas price for all types of conversion currencies. 
Removed unnecessary external library for currency formatting since it is now natively supported by ECMAScript Internationalization API.

**Checklist**

* [x] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**

Resolves #3498

### CODE REVIEW

**Review by andreahaku — APPROVED** [review]
LGTM

**Review by cortisiko — CHANGES_REQUESTED** [review]
[Android specific] If i were to go:

change my fiat currency from USD to BNT
go to uniswap 
attempt to swap, the app crashes: http://recordit.co/knzyiMM8nw 

I see this in the logs: 

```
 ERROR  ReferenceError: Can't find variable: Intl

This error is located at:
    in TransactionReviewInformation (created by Connect(TransactionReviewInformation))
    in Connect(TransactionReviewInformation) (at TransactionReview/index.js:376)
    in RCTView (at View.js:32)
    in View (at TransactionReview/index.js:372)
    in TouchableWithoutFeedback (at ActionView/index.js:54)
    in RCTView (at View.js:32)
    in View (at ScrollView.js:1674)
    in RCTScrollView (at ScrollView.js:1792)
    in ScrollView (at ScrollView.js:1818)
    in ScrollView (at KeyboardAwareHOC.js:537)
    in KeyboardAwareScrollView (at ActionView/index.js:49)
    in RCTView (at View.js:32)
    in View (at ActionView/index.js:48)
    in ActionView (at TransactionReview/index.js:364)
    in RCTView (at View.js:32)
    in View (at TransactionReview/index.js:363)
    in RCTView (at View.js:32)
    in View (at createAnimatedComponent.js:242)
    in AnimatedComponent (at createAnimatedComponent.js:295)
    in AnimatedComponentWrapper (at TransactionReview/index.js:353)
    in TransactionReview (created by Connect(TransactionReview))
    in Connect(TransactionReview) (at TransactionEditor/index.js:809)
    in RCTView (at View.js:32)
    in View (at createAnimatedComponent.js:242)
    in AnimatedComponent (at createAnimatedComponent.js:295)
    in AnimatedComponentWrapper (at AnimatedTransactionModal/index.js:225)
    in RCTView (at View.js:32)
    in View (at createAnimatedComponent.js:242)
    in AnimatedComponent (at createAnimatedComponent.js:295)
    in AnimatedComponentWrapper (at AnimatedTransactionModal/index.js:216)
    in AnimatedTransactionModal (at TransactionEditor/index.js:808)
    in RCTView (at View.js:32)
    in View (at ScrollView.js:1674)
    in RCTScrollView (at ScrollView.js:1792)
    in ScrollView (at ScrollView.js:1818)
    in ScrollView (at KeyboardAwareHOC.js:537)
    in KeyboardAwareScrollView (at TransactionEditor/index.js:807)
    in TransactionEditor (created by Connect(TransactionEditor))
    in Connect(TransactionEditor) (at Approval/index.js:361)
    in RCTView (at View.js:32)
    in View (at createAnimatedComponent.js:242)
    in AnimatedComponent (at createAnimatedComponent.js:295)
    in AnimatedComponentWrapper (at createAnimatableComponent.js:599)
    in withAnimatable(View) (created by ReactNativeModal)
    in RCTView (at View.js:32)
    in View (at AppContainer.js:92)
    in RCTView (at View.js:32)
    in View (at AppContainer.js:119)
    in AppContainer (at Modal.js:242)
    in RCTView (at View.js:32)
    in View (at Modal.js:270)
    in VirtualizedListContextResetter (at Modal.js:268)
    in RCTModalHostView (at Modal.js:248)
    in Modal (created by ReactNativeModal)
    in ReactNativeModal (at Approval/index.js:347)
    in Approval (created by Connect(Approval))
    in Connect(Approval) (at Main/index.js:547)
    in Main (created by Connect(Main))
    in Connect(Main) (at SceneView.tsx:122)
    in StaticContainer
    in EnsureSingleNavigator (at SceneView.tsx:114)
    in SceneView (at useDescriptors.tsx:153)
    in RCTView (at View.js:32)
    in View (at CardContainer.tsx:245)
    in RCTView (at View.js:32)
    in View (at CardContainer.tsx:244)
    in RCTView (at View.js:32)
    in View (at CardSheet.tsx:33)
    in CardSheet (at Card.tsx:573)
    in RCTView (at View.js:32)
    in View (at createAnimatedComponent.js:242)
    in AnimatedComponent (at createAnimatedComponent.js:295)
    in AnimatedComponentWrapper (at Card.tsx:555)
    in PanGestureHandler (at GestureHandlerNative.tsx:13)
    in PanGestureHandler (at Card.tsx:549)
    in RCTView (at View.js:32)
    in View (at createAnimatedComponent.js:242)
    in AnimatedComponent (at createAnimatedComponent.js:295)
    in AnimatedComponentWrapper (at Card.tsx:544)
    in RCTView (at View.js:32)
    in View (at Card.tsx:538)
    in Card (at CardContainer.tsx:206)
    in CardContainer (at CardStack.tsx:623)
    in RNSScreen (at createAnimatedComponent.js:242)
    in AnimatedComponent (at createAnimatedComponent.js:295)
    in AnimatedComponentWrapper (at src/index.native.tsx:147)
    in Screen (at Screens.tsx:74)
    in MaybeScreen (at CardStack.tsx:616)
    in RNSScreenContainer (at src/index.native.tsx:186)
    in ScreenContainer (at Screens.tsx:50)
    in MaybeScreenContainer (at CardStack.tsx:498)
    in CardStack (at StackView.tsx:462)
    in KeyboardManager (at StackView.tsx:458)
    in SafeAreaProviderCompat (at StackView.tsx:455)
    in RCTView (at View.js:32)
    in View (at GestureHandlerRootView.android.tsx:21)
    in GestureHandlerRootView (at StackView.tsx:454)
    in StackView (at createStackNavigator.tsx:87)
    in StackNavigator (at Main/index.js:809)
    in MainFlow (at SceneView.tsx:122)
    in StaticContainer
    in EnsureSingleNavigator (at SceneView.tsx:114)
    in SceneView (at useDescriptors.tsx:153)
    in RCTView (at View.js:32)
    in View (at CardContainer.tsx:245)
    in RCTView (at View.js:32)
    in View (at CardContainer.tsx:244)
    in RCTView (at View.js:32)
    in View (at CardSheet.tsx:33)
    in CardSheet (at Card.tsx:573)
    in RCTView (at View.js:32)
    in View (at createAnimatedComponent.js:242)
    in AnimatedComponent (at createAnimatedComponent.js:295)
    in AnimatedComponentWrapper (at Card.tsx:555)
    in PanGestureHandler (at GestureHandlerNative.tsx:13)
    in PanGestureHandler (at Card.tsx:549)
    in RCTView (at View.js:32)
    in View (at createAnimatedComponent.js:242)
    in AnimatedComponent (at createAnimatedComponent.js:295)
    in AnimatedComponentWrapper (at Card.tsx:544)
    in RCTView (at View.js:32)
    in View (at Card.tsx:538)
    in Card (at CardContainer.tsx:206)
    in CardContainer (at CardStack.tsx:623)
    in RNSScreen (at createAnimatedComponent.js:242)
    in AnimatedComponent (at createAnimatedComponent.js:295)
    in AnimatedComponentWrapper (at src/index.native.tsx:147)
    in Screen (at Screens.tsx:74)
    in MaybeScreen (at CardStack.tsx:616)
    in RNSScreenContainer (at src/index.native.tsx:186)
    in ScreenContainer (at Screens.tsx:50)
    in MaybeScreenContainer (at CardStack.tsx:498)
    in CardStack (at StackView.tsx:462)
    in KeyboardManager (at StackView.tsx:458)
    in RNCSafeAreaProvider (at SafeAreaContext.tsx:76)
    in SafeAreaProvider (at SafeAreaProviderCompat.tsx:42)
    in SafeAreaProviderCompat (at StackView.tsx:455)
    in GestureHandlerRootView (at GestureHandlerRootView.android.tsx:26)
    in GestureHandlerRootView (at StackView.tsx:454)
    in StackView (at createStackNavigator.tsx:87)
    in StackNavigator (at App/index.js:277)
    in EnsureSingleNavigator (at BaseNavigationContainer.tsx:409)
    in BaseNavigationContainer (at NavigationContainer.tsx:91)
    in ThemeProvider (at NavigationContainer.tsx:90)
    in NavigationContainer (at App/index.js:266)
    in RCTView (at View.js:32)
    in View (at App/index.js:265)
    in App (created by Connect(App))
    in Connect(App) (at Root/index.js:44)
    in ErrorBoundary (at Root/index.js:43)
    in PersistGate (at Root/index.js:42)
    in Provider (at Root/index.js:41)
    in Root (at renderApplication.js:50)
    in RCTView (at View.js:32)
    in View (at AppContainer.js:92)
    in RCTView (at View.js:32)
    in View (at AppContainer.js:119)
    in AppContainer (at renderApplication.js:43)
    in MetaMask(RootComponent) (at renderApplication.js:60)
```

File: app/util/confirm-tx.js [hunk 1]
```diff
@@ -86,13 +85,13 @@ export function getTransactionFee({ value, fromCurrency = 'ETH', toCurrency, con
...
 	const upperCaseCurrencyCode = currencyCode.toUpperCase();
 
-	return currencies.find((currency) => currency.code === upperCaseCurrencyCode)
-		? currencyFormatter.format(Number(value), {
-				code: upperCaseCurrencyCode,
-				style: 'currency',
-				// eslint-disable-next-line no-mixed-spaces-and-tabs
-		  })
-		: `value ${upperCaseCurrencyCode}`;
+	const formatedCurrency = new Intl.NumberFormat(I18n.locale, {
```
Comment (andreahaku) [inline app/util/confirm-tx.js:88]: to support `Intl` on Android you should modify `build.gradle` like that:
``` /**
   * Use the international variant of JavaScriptCore
   * This variant includes the ICU i18n library to make APIs like `Date.toLocaleString`
   * and `String.localeCompare` work when using with locales other than en-US.
   * Note that this variant is about 6MiB larger per architecture than the default.
   */
-  def useIntlJsc = false
+  def useIntlJsc = true
```

Comment (Fatxx) [reply app/util/confirm-tx.js:88]: Thanks @andreahaku

**Comment by Fatxx** [comment]
@cortisiko just fixed you can QA again if you like

**Review by cortisiko — APPROVED** [review]
🌮 🌮 🌮 🌮

**Pull request merged**